NullPointerException when using the bnd-export-maven-plugin behind a proxy

When using the `bnd-export-maven-plugin`(version 3.4.0) behind a proxy I see a NullPointerException

    Caused by: java.lang.NullPointerException
        at aQute.bnd.repository.osgi.OSGiIndex.download(OSGiIndex.java:86)
        at aQute.bnd.repository.osgi.OSGiIndex.readIndexes(OSGiIndex.java:58)
        at aQute.bnd.repository.osgi.OSGiIndex.<init>(OSGiIndex.java:51)
        at aQute.bnd.repository.osgi.OSGiRepository.getIndex(OSGiRepository.java:156)
        at aQute.bnd.repository.osgi.OSGiRepository.getIndex(OSGiRepository.java:106)
        at aQute.bnd.repository.osgi.OSGiRepository.prepare(OSGiRepository.java:302)
        at aQute.bnd.build.Workspace.getRepositories(Workspace.java:535)
        at aQute.bnd.maven.export.plugin.ExportMojo.export(ExportMojo.java:110)
        at aQute.bnd.maven.export.plugin.ExportMojo.execute(ExportMojo.java:84)
        ...

Note that I see the same exception when using Bndtools 3.4.0 from behind the proxy, but with the following stack:

    Caused by: java.lang.NullPointerException
        at aQute.bnd.repository.osgi.OSGiIndex.download(OSGiIndex.java:86)
        at aQute.bnd.repository.osgi.OSGiIndex.readIndexes(OSGiIndex.java:58)
        at aQute.bnd.repository.osgi.OSGiIndex.<init>(OSGiIndex.java:51)
        at aQute.bnd.repository.osgi.OSGiRepository.getIndex(OSGiRepository.java:156)
        at aQute.bnd.repository.osgi.OSGiRepository.getIndex(OSGiRepository.java:106)
        at aQute.bnd.repository.osgi.OSGiRepository.list(OSGiRepository.java:186)
        at bndtools.model.repo.RepositoryTreeContentProvider$1.run(RepositoryTreeContentProvider.java:293)
        at org.eclipse.core.internal.jobs.Worker.run(Worker.java:55)

The proxy is configured in `~/.m2/settings.xml`. When the proxy is commented out the build completes successfully.

--- a/biz.aQute.bndlib.comm.tests/bnd.bnd
+++ b/biz.aQute.bndlib.comm.tests/bnd.bnd
@@ -18,5 +18,7 @@
 	org.apache.commons.lang3;version=latest,\
 	org.apache.commons.codec;version=latest,\
 	biz.aQute.repository;version=latest,\
-	com.google.guava;version=latest
+	com.google.guava;version=latest,\
+	bcpkix;version=latest,\
+	bcprov;version=latest
 

--- a/biz.aQute.bndlib.comm.tests/test/aQute/bnd/comm/tests/HttpClientProxyTest.java
+++ b/biz.aQute.bndlib.comm.tests/test/aQute/bnd/comm/tests/HttpClientProxyTest.java
@@ -1,15 +1,38 @@
 package aQute.bnd.comm.tests;
 
 import java.io.IOException;
+import java.math.BigInteger;
 import java.net.MalformedURLException;
 import java.net.Proxy.Type;
+import java.net.URI;
 import java.net.URL;
+import java.security.KeyPair;
+import java.security.KeyPairGenerator;
+import java.security.NoSuchAlgorithmException;
+import java.security.SecureRandom;
+import java.security.cert.X509Certificate;
+import java.util.Date;
 import java.util.concurrent.atomic.AtomicBoolean;
 import java.util.concurrent.atomic.AtomicInteger;
 import java.util.concurrent.atomic.AtomicReference;
 
+import javax.net.ssl.SSLEngine;
+import javax.net.ssl.SSLException;
+import javax.net.ssl.SSLSession;
+
+import org.bouncycastle.asn1.x500.X500NameBuilder;
+import org.bouncycastle.asn1.x500.style.BCStyle;
+import org.bouncycastle.cert.X509v3CertificateBuilder;
+import org.bouncycastle.cert.jcajce.JcaX509CertificateConverter;
+import org.bouncycastle.cert.jcajce.JcaX509v3CertificateBuilder;
+import org.bouncycastle.operator.ContentSigner;
+import org.bouncycastle.operator.jcajce.JcaContentSignerBuilder;
+import org.littleshoot.proxy.HttpFilters;
+import org.littleshoot.proxy.HttpFiltersAdapter;
+import org.littleshoot.proxy.HttpFiltersSource;
 import org.littleshoot.proxy.HttpProxyServer;
 import org.littleshoot.proxy.HttpProxyServerBootstrap;
+import org.littleshoot.proxy.MitmManager;
 import org.littleshoot.proxy.ProxyAuthenticator;
 import org.littleshoot.proxy.impl.DefaultHttpProxyServer;
 
@@ -24,6 +47,15 @@ import aQute.http.testservers.HttpTestServer;
 import aQute.http.testservers.Httpbin;
 import aQute.lib.io.IO;
 import aQute.lib.strings.Strings;
+import io.netty.buffer.UnpooledByteBufAllocator;
+import io.netty.channel.ChannelHandlerContext;
+import io.netty.handler.codec.http.DefaultHttpResponse;
+import io.netty.handler.codec.http.HttpObject;
+import io.netty.handler.codec.http.HttpRequest;
+import io.netty.handler.codec.http.HttpResponse;
+import io.netty.handler.codec.http.HttpResponseStatus;
+import io.netty.handler.codec.http.HttpVersion;
+import io.netty.handler.ssl.SslContextBuilder;
 import junit.framework.TestCase;
 import sockslib.common.AuthenticationException;
 import sockslib.common.Credentials;
@@ -45,6 +77,50 @@ import sockslib.server.msg.CommandMessage;
  * https://github.com/fengyouchao/sockslib for socks
  */
 public class HttpClientProxyTest extends TestCase {
+
+	private final class ProxyCheckingFilter implements HttpFiltersSource {
+
+		private boolean secure;
+
+		public ProxyCheckingFilter(boolean secure) {
+			this.secure = secure;
+		}
+
+		@Override
+		public int getMaximumResponseBufferSizeInBytes() {
+			return 0;
+		}
+
+		@Override
+		public int getMaximumRequestBufferSizeInBytes() {
+			return 0;
+		}
+
+		@Override
+		public HttpFilters filterRequest(HttpRequest arg0, ChannelHandlerContext arg1) {
+			return new HttpFiltersAdapter(arg0, arg1) {
+
+				@Override
+				public HttpResponse clientToProxyRequest(HttpObject obj) {
+
+					String scheme = URI.create(originalRequest.uri()).getScheme();
+
+					if (!secure && "http".equalsIgnoreCase(scheme)) {
+						// Insecure using HTTP
+						proxyCalled.set(true);
+						return null;
+					} else if (secure && !"http".equalsIgnoreCase(scheme)) {
+						// The scheme gets eaten when we use SSL, so we just
+						// have to check that it's not HTTP
+						proxyCalled.set(true);
+						return null;
+					} else {
+						return new DefaultHttpResponse(HttpVersion.HTTP_1_1, HttpResponseStatus.BAD_REQUEST);
+					}
+				}
+			};
+		}
+	}
 
 	public void testPromiscuousProxyWithNoUser() throws Exception {
 		createPromiscuousHttpProxy();
@@ -73,18 +149,18 @@ public class HttpClientProxyTest extends TestCase {
 
 	public void testNoProxyUnsecure() throws Exception {
 		createUnsecureServer();
-		assertHttpProxy("good", false);
+		assertHttpProxy("good", Type.HTTP.name(), false, false);
 	}
 
 	public void testNoProxySecure() throws Exception {
 		createUnsecureServer();
-		assertHttpProxy("good", false);
+		assertHttpProxy("good", Type.HTTP.name(), false, false);
 	}
 
 	public void testPromiscuousProxyWithGoodUserSecure() throws Exception {
-		createPromiscuousHttpProxy();
 		createSecureServer();
-		assertHttpProxy("good", false);
+		createSecurePromiscuousHttpProxy();
+		assertHttpProxy("good", "HTTPS", true, false);
 	}
 
 	public void testAuthenticatingProxyNoUser() throws Exception {
@@ -151,6 +227,7 @@ public class HttpClientProxyTest extends TestCase {
 
 	private HttpProxyServer				httpProxy;
 	private AtomicBoolean				authenticationCalled	= new AtomicBoolean();
+	private AtomicBoolean				proxyCalled				= new AtomicBoolean();
 	private HttpTestServer				httpTestServer;
 	private SocksProxyServer			socks5Proxy;
 	private static int					httpProxyPort			= 2080;
@@ -161,22 +238,102 @@ public class HttpClientProxyTest extends TestCase {
 	// we use different ports because the servers seem to linger
 
 	void createAuthenticationHttpProxy() {
-		HttpProxyServerBootstrap bootstrap = DefaultHttpProxyServer.bootstrap().withPort(++httpProxyPort);
-		bootstrap.withProxyAuthenticator(new ProxyAuthenticator() {
+		HttpProxyServerBootstrap bootstrap = DefaultHttpProxyServer.bootstrap()
+				.withPort(++httpProxyPort)
+				.withProxyAuthenticator(new ProxyAuthenticator() {
 
-			@Override
-			public boolean authenticate(String user, String password) {
-				System.out.println("Authenticating " + user + " : " + password);
-				authenticationCalled.set(true);
-				return "proxyuser".equals(user) && "good".equals(password);
+					@Override
+					public boolean authenticate(String user, String password) {
+						System.out.println("Authenticating " + user + " : " + password);
+						authenticationCalled.set(true);
+						return "proxyuser".equals(user) && "good".equals(password);
 
-			}
-		});
+					}
+
+					@Override
+					public String getRealm() {
+						return null;
+					}
+				})
+				.withFiltersSource(new ProxyCheckingFilter(false));
 		httpProxy = bootstrap.start();
 	}
 
 	void createPromiscuousHttpProxy() {
-		HttpProxyServerBootstrap bootstrap = DefaultHttpProxyServer.bootstrap().withPort(++httpProxyPort);
+		HttpProxyServerBootstrap bootstrap = DefaultHttpProxyServer.bootstrap()
+				.withPort(++httpProxyPort)
+				.withFiltersSource(new ProxyCheckingFilter(false));
+		httpProxy = bootstrap.start();
+	}
+
+	void createSecurePromiscuousHttpProxy() {
+		HttpProxyServerBootstrap bootstrap = DefaultHttpProxyServer.bootstrap()
+				.withPort(++httpProxyPort)
+				.withManInTheMiddle(new MitmManager() {
+
+					SecureRandom random = new SecureRandom();
+
+					@Override
+					public SSLEngine serverSslEngine(String arg0, int arg1) {
+						return serverSslEngine();
+					}
+
+					@Override
+					public SSLEngine serverSslEngine() {
+						try {
+							return SslContextBuilder.forClient()
+									.trustManager(httpTestServer.getCertificateChain())
+									.build()
+									.newEngine(UnpooledByteBufAllocator.DEFAULT);
+						} catch (SSLException e) {
+							throw new RuntimeException(e);
+						}
+					}
+
+					@SuppressWarnings("restriction")
+					private X509Certificate[] createSelfSignedCertifcate(KeyPair keyPair) throws Exception {
+						X500NameBuilder nameBuilder = new X500NameBuilder(BCStyle.INSTANCE);
+						nameBuilder.addRDN(BCStyle.CN, "localhost");
+
+						Date notBefore = new Date();
+						Date notAfter = new Date(System.currentTimeMillis() + 24 * 3 * 60 * 60 * 1000);
+
+						BigInteger serialNumber = new BigInteger(128, random);
+
+						X509v3CertificateBuilder certificateBuilder = new JcaX509v3CertificateBuilder(
+								nameBuilder.build(), serialNumber, notBefore, notAfter, nameBuilder.build(),
+								keyPair.getPublic());
+						ContentSigner contentSigner = new JcaContentSignerBuilder("SHA256WithRSAEncryption")
+								.build(keyPair.getPrivate());
+						X509Certificate certificate = new JcaX509CertificateConverter()
+								.getCertificate(certificateBuilder.build(contentSigner));
+						return new X509Certificate[] {
+								certificate
+						};
+					}
+
+					private KeyPair createKey() throws NoSuchAlgorithmException {
+						KeyPairGenerator keyGen = KeyPairGenerator.getInstance("RSA");
+						keyGen.initialize(2048, random);
+						KeyPair pair = keyGen.generateKeyPair();
+						return pair;
+					}
+
+					@Override
+					public SSLEngine clientSslEngineFor(HttpRequest arg0, SSLSession arg1) {
+						try {
+							KeyPair keyPair = createKey();
+							X509Certificate[] selfSignedCertifcate = createSelfSignedCertifcate(keyPair);
+
+							return SslContextBuilder.forServer(keyPair.getPrivate(), selfSignedCertifcate)
+									.build()
+									.newEngine(UnpooledByteBufAllocator.DEFAULT);
+						} catch (Exception e) {
+							throw new RuntimeException(e);
+						}
+					}
+				})
+				.withFiltersSource(new ProxyCheckingFilter(true));
 		httpProxy = bootstrap.start();
 	}
 
@@ -197,18 +354,19 @@ public class HttpClientProxyTest extends TestCase {
 	private void createSecureSocks5() throws IOException, InterruptedException {
 		UserManager userManager = new MemoryBasedUserManager();
 		userManager.create(new User("proxyuser", "good"));
-		SocksServerBuilder builder = SocksServerBuilder.newSocks5ServerBuilder();
-		builder.setBindPort(++socksProxyPort);
-		builder.setUserManager(userManager);
-		builder.addSocksMethods(new UsernamePasswordMethod(new UsernamePasswordAuthenticator() {
-			@Override
-			public void doAuthenticate(Credentials arg0, Session arg1) throws AuthenticationException {
-				System.out.println("Authenticating"); // does not get called?
-				super.doAuthenticate(arg0, arg1);
-				authenticationCalled.set(true);
+		SocksServerBuilder builder = SocksServerBuilder.newSocks5ServerBuilder()
+				.setBindPort(++socksProxyPort)
+				.setUserManager(userManager)
+				.addSocksMethods(new UsernamePasswordMethod(new UsernamePasswordAuthenticator() {
+					@Override
+					public void doAuthenticate(Credentials arg0, Session arg1) throws AuthenticationException {
+						System.out.println("Authenticating"); // does not get
+																// called?
+						super.doAuthenticate(arg0, arg1);
+						authenticationCalled.set(true);
 
-			}
-		}));
+					}
+				}));
 		socks5Proxy = builder.build();
 
 		socks5Proxy.getSessionManager().addSessionListener("abc", new SessionListener() {
@@ -222,6 +380,7 @@ public class HttpClientProxyTest extends TestCase {
 
 			@Override
 			public void onCommand(Session arg0, CommandMessage arg1) throws CloseSessionException {
+				proxyCalled.set(true);
 				System.err.println("Command " + arg0 + " " + arg1);
 			}
 
@@ -243,6 +402,10 @@ public class HttpClientProxyTest extends TestCase {
 	@Override
 	protected void tearDown() throws Exception {
 		super.tearDown();
+
+		proxyCalled.set(false);
+		authenticationCalled.set(false);
+
 		if (httpProxy != null)
 			httpProxy.abort();
 
@@ -253,8 +416,13 @@ public class HttpClientProxyTest extends TestCase {
 			socks5Proxy.shutdown();
 	}
 
-	@SuppressWarnings("resource")
 	void assertHttpProxy(String password, boolean authenticationCalled) throws MalformedURLException, Exception {
+		assertHttpProxy(password, Type.HTTP.name(), true, authenticationCalled);
+	}
+
+	@SuppressWarnings("resource")
+	void assertHttpProxy(String password, String protocol, boolean proxyCalled, boolean authenticationCalled)
+			throws MalformedURLException, Exception {
 		Processor p = new Processor();
 		p.setProperty("-connectionsettings", "" + false);
 		HttpClient hc = new HttpClient();
@@ -265,12 +433,16 @@ public class HttpClientProxyTest extends TestCase {
 			proxy.active = true;
 			proxy.host = "localhost";
 			proxy.port = httpProxyPort;
-			proxy.protocol = Type.HTTP;
+			proxy.protocol = protocol;
 			if (password != null) {
 				proxy.username = "proxyuser";
 				proxy.password = password;
 			}
 			hc.addProxyHandler(ConnectionSettings.createProxyHandler(proxy));
+
+			if ("HTTPS".equals(protocol)) {
+
+			}
 		}
 
 		ServerDTO server = new ServerDTO();
@@ -287,7 +459,8 @@ public class HttpClientProxyTest extends TestCase {
 		String s = IO.collect(tag.getInputStream());
 		assertNotNull(s);
 		assertTrue(s.trim().startsWith("{"));
-		assertTrue(this.authenticationCalled.get() == authenticationCalled);
+		assertEquals(proxyCalled, this.proxyCalled.get());
+		assertEquals(authenticationCalled, this.authenticationCalled.get());
 
 	}
 
@@ -304,7 +477,7 @@ public class HttpClientProxyTest extends TestCase {
 			proxy.active = true;
 			proxy.host = "localhost";
 			proxy.port = socksProxyPort;
-			proxy.protocol = Type.SOCKS;
+			proxy.protocol = Type.SOCKS.name();
 			if (password != null) {
 				proxy.username = "proxyuser";
 				proxy.password = password;
@@ -325,6 +498,7 @@ public class HttpClientProxyTest extends TestCase {
 		String s = IO.collect(tag.getInputStream());
 		assertNotNull(s);
 		assertTrue(s.trim().startsWith("{"));
+		assertTrue(proxyCalled.get());
 		// assertTrue(this.authenticationCalled.get() == authenticationCalled);
 
 	}

--- a/biz.aQute.bndlib.comm.tests/test/aQute/bnd/comm/tests/SettingsParserTest.java
+++ b/biz.aQute.bndlib.comm.tests/test/aQute/bnd/comm/tests/SettingsParserTest.java
@@ -51,13 +51,37 @@ public class SettingsParserTest extends TestCase {
 		assertEquals("passwd", p.password);
 	}
 
+	public void testProxies() throws Exception {
+		SettingsDTO settings = getSettings("proxy-types.xml");
+		assertEquals(2, settings.proxies.size());
+		ProxyDTO p = settings.proxies.get(0);
+		assertEquals("http-proxy", p.id);
+		assertEquals(true, p.active);
+		assertEquals(Type.HTTP.name(), p.protocol.toUpperCase());
+		assertEquals("localhost", p.host);
+		assertEquals(80, p.port);
+		assertEquals(null, p.nonProxyHosts);
+		assertEquals(null, p.username);
+		assertEquals(null, p.password);
+
+		p = settings.proxies.get(1);
+		assertEquals("https-proxy", p.id);
+		assertEquals(true, p.active);
+		assertEquals("HTTPS", p.protocol.toUpperCase());
+		assertEquals("localhost", p.host);
+		assertEquals(443, p.port);
+		assertEquals(null, p.nonProxyHosts);
+		assertEquals(null, p.username);
+		assertEquals(null, p.password);
+	}
+
 	public void testSocksAuth() throws Exception {
 		SettingsDTO settings = getSettings("socks-auth.xml");
 		assertEquals(1, settings.proxies.size());
 		ProxyDTO p = settings.proxies.get(0);
 		assertEquals("myproxy", p.id);
 		assertEquals(true, p.active);
-		assertEquals(Type.SOCKS, p.protocol);
+		assertEquals(Type.SOCKS.name(), p.protocol.toUpperCase());
 		assertEquals(1080, p.port);
 		assertEquals(null, p.nonProxyHosts);
 		assertEquals("proxyuser", p.username);
@@ -70,7 +94,7 @@ public class SettingsParserTest extends TestCase {
 		ProxyDTO p = settings.proxies.get(0);
 		assertEquals("myproxy", p.id);
 		assertEquals(true, p.active);
-		assertEquals(Type.SOCKS, p.protocol);
+		assertEquals(Type.SOCKS.name(), p.protocol.toUpperCase());
 		assertEquals(1080, p.port);
 		assertEquals(null, p.nonProxyHosts);
 		assertEquals(null, p.username);
@@ -83,7 +107,7 @@ public class SettingsParserTest extends TestCase {
 		ProxyDTO p = settings.proxies.get(0);
 		assertEquals("myproxy", p.id);
 		assertEquals(true, p.active);
-		assertEquals(Type.SOCKS, p.protocol);
+		assertEquals(Type.SOCKS.name(), p.protocol.toUpperCase());
 		assertEquals(1080, p.port);
 		assertEquals("*.google.com|ibiblio.org", p.nonProxyHosts);
 		assertEquals(null, p.username);

--- a/biz.aQute.bndlib.comm.tests/test/aQute/bnd/comm/tests/Standalone.java
+++ b/biz.aQute.bndlib.comm.tests/test/aQute/bnd/comm/tests/Standalone.java
@@ -79,6 +79,12 @@ public class Standalone {
 				return "proxyuser".equals(user) && "good".equals(password);
 
 			}
+
+			@Override
+			public String getRealm() {
+				// TODO Auto-generated method stub
+				return null;
+			}
 		});
 		bootstrap.start();
 	}

--- a/biz.aQute.bndlib.comm.tests/testresources/proxy-types.xml
+++ b/biz.aQute.bndlib.comm.tests/testresources/proxy-types.xml
@@ -1,0 +1,31 @@
+
+<settings xmlns="http://maven.apache.org/SETTINGS/1.0.0"
+	xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+	xsi:schemaLocation="http://maven.apache.org/SETTINGS/1.0.0
+                          http://maven.apache.org/xsd/settings-1.0.0.xsd">
+	<localRepository />
+	<interactiveMode />
+	<usePluginRegistry />
+	<offline />
+	<pluginGroups />
+	<servers />
+	<mirrors />
+	<proxies>
+		<proxy>
+			<id>http-proxy</id>
+			<active>true</active>
+			<protocol>http</protocol>
+			<host>localhost</host>
+			<port>80</port>
+		</proxy>
+		<proxy>
+			<id>https-proxy</id>
+			<active>true</active>
+			<protocol>https</protocol>
+			<host>localhost</host>
+			<port>443</port>
+		</proxy>
+	</proxies>
+	<profiles />
+	<activeProfiles />
+</settings>

--- a/biz.aQute.bndlib/src/aQute/bnd/connection/settings/ConnectionSettings.java
+++ b/biz.aQute.bndlib/src/aQute/bnd/connection/settings/ConnectionSettings.java
@@ -7,6 +7,7 @@ import java.net.InterfaceAddress;
 import java.net.NetworkInterface;
 import java.net.PasswordAuthentication;
 import java.net.Proxy;
+import java.net.Proxy.Type;
 import java.net.SocketAddress;
 import java.net.SocketException;
 import java.net.URL;
@@ -241,24 +242,37 @@ public class ConnectionSettings extends Processor {
 
 			@Override
 			public ProxySetup forURL(URL url) throws Exception {
-				switch (proxyDTO.protocol) {
 
-					case DIRECT :
+				Proxy.Type type;
+
+				switch (proxyDTO.protocol.toUpperCase()) {
+
+					case "DIRECT" :
+						type = Type.DIRECT;
 						break;
 
-					case HTTP :
-						String scheme = url.getProtocol();
-						if (scheme.equalsIgnoreCase("http") || scheme.equalsIgnoreCase("https")) {
+					case "HTTP" :
+						type = Type.HTTP;
+						if (url.getProtocol().equalsIgnoreCase("http")) {
 							// ok
 						} else
 							return null;
 
 						break;
+					case "HTTPS" :
+						type = Type.HTTP;
+						if (url.getProtocol().equalsIgnoreCase("https")) {
+							// ok
+						} else
+							return null;
+						break;
 
-					case SOCKS :
+					case "SOCKS" :
+						type = Type.SOCKS;
 						break;
 
 					default :
+						type = Type.HTTP;
 						break;
 				}
 
@@ -283,7 +297,7 @@ public class ConnectionSettings extends Processor {
 					else
 						socketAddress = new InetSocketAddress(proxyDTO.port);
 
-					proxySetup.proxy = new Proxy(proxyDTO.protocol, socketAddress);
+					proxySetup.proxy = new Proxy(type, socketAddress);
 				}
 				return proxySetup;
 			}

--- a/biz.aQute.bndlib/src/aQute/bnd/connection/settings/ProxyDTO.java
+++ b/biz.aQute.bndlib/src/aQute/bnd/connection/settings/ProxyDTO.java
@@ -10,7 +10,7 @@ public class ProxyDTO extends DTO {
 	public String	id			= "default";
 	public boolean	active		= true;
 	public String	mask;
-	public Type		protocol	= Type.HTTP;
+	public String	protocol	= Type.HTTP.name();
 	public String	username;
 	public String	password;
 	public int		port		= 8080;

--- a/cnf/ext/central.mvn
+++ b/cnf/ext/central.mvn
@@ -78,7 +78,7 @@ org.slf4j:jcl-over-slf4j:1.7.13
 
 org.yaml:snakeyaml:1.15
 
-org.littleshoot:littleproxy:1.1.0-beta1
+org.littleshoot:littleproxy:1.1.2
 
 io.netty:netty-all:4.1.0.Beta8
 

